plugin/src/main/java/org/gradle/testretry/internal/executer/TestFrameworkTemplate.java

```diff
@@ -20,18 +20,21 @@
 import org.gradle.internal.reflect.Instantiator;
 import org.gradle.testretry.internal.testsreader.TestsReader;
 
+import java.io.File;
+import java.util.Set;
+
 public class TestFrameworkTemplate {
 
     public final Test task;
     public final Instantiator instantiator;
     public final ObjectFactory objectFactory;
     public final TestsReader testsReader;
 
-    public TestFrameworkTemplate(Test task, Instantiator instantiator, ObjectFactory objectFactory) {
+    public TestFrameworkTemplate(Test task, Instantiator instantiator, ObjectFactory objectFactory, Set<File> testClassesDir, Set<File> resolvedClasspath) {
         this.task = task;
         this.instantiator = instantiator;
         this.objectFactory = objectFactory;
-        this.testsReader = new TestsReader(task.getTestClassesDirs().getFiles(), task.getClasspath());
+        this.testsReader = new TestsReader(testClassesDir, resolvedClasspath);
```

Comment:
Why was this triggering resolution? Shouldn't these have already been resolved prior to executing the task?

Reply:
The lock is required before attempting to resolve, i.e. before the “has this already been resolved?” check happens.